py_console_script_binary fails mypy [func-returns-value]

# 🐞 bug report

### Affected Rule

The issue is caused by the rule: `py_console_script_binary`


### Is this a regression?

I don't believe so.

### Description

When running mypy on a `py_console_script_binary` wrapping a pip package (in this case, [black](https://pypi.org/project/black/)), I get an error:

```
bazel-out/k8-fastbuild/bin/[path in my repo]/rules_python_entry_point_black.py:19: error: "patched_main" does not return a value (it only ever returns None)  [func-returns-value]
```

The issue is basically that the `{entry_point}` being wrapped [here](https://github.com/bazel-contrib/rules_python/blob/ab3e3f790788e119bfb362a60819727d7450f0f3/python/private/py_console_script_gen.py#L65) may not have a return value.

## 🔬 Minimal Reproduction

I don't know if this is minimal enough for you, but:

```
git clone https://pigweed.googlesource.com/pigweed/pigweed
cd pigweed
git fetch https://pigweed.googlesource.com/pigweed/pigweed refs/changes/73/309473/1 && git checkout -b change-309473 FETCH_HEAD
bazelisk build --config=mypy //pw_presubmit/py:black
```

## 🔥 Exception or Error

```
bazel-out/k8-fastbuild/bin/[path in my repo]/rules_python_entry_point_black.py:19: error: "patched_main" does not return a value (it only ever returns None)  [func-returns-value]
```

</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Linux
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
8.1.0
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
0.40.0
  </code>
</pre>

**Anything else relevant?**

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -65,6 +65,8 @@ END_UNRELEASED_TEMPLATE
 ### Fixed
 * (gazelle) Fixed handling of auto-included `__init__.py` files when generating `py_binary`
   targets ([#3729](https://github.com/bazel-contrib/rules_python/issues/3729)).
+* (entry_point) From now on `mypy` type checking will be skipped on the generated
+  files ([#3126](https://github.com/bazel-contrib/rules_python/issues/3126)).
 
 {#v0-0-0-added}
 ### Added

--- a/python/private/py_console_script_gen.py
+++ b/python/private/py_console_script_gen.py
@@ -62,6 +62,7 @@ except ImportError:
     raise
 
 if __name__ == "__main__":
+    # type: ignore
     sys.exit({entry_point}())
 """
 

--- a/python/private/py_console_script_gen.py
+++ b/python/private/py_console_script_gen.py
@@ -62,8 +62,7 @@ except ImportError:
     raise
 
 if __name__ == "__main__":
-    # type: ignore
-    sys.exit({entry_point}())
+    sys.exit({entry_point}())  # type: ignore
 """
 
 

--- a/tests/entry_points/py_console_script_gen_test.py
+++ b/tests/entry_points/py_console_script_gen_test.py
@@ -162,7 +162,7 @@ class RunTest(unittest.TestCase):
             raise
 
         if __name__ == "__main__":
-            sys.exit(baz())
+            sys.exit(baz())  # type: ignore
         """
         )
         self.assertEqual(want, got)

